examples/searching/Binary Search/Go/recursive.go

```diff
@@ -0,0 +1,76 @@
+/**
```

Comment:
See https://github.com/johnfraserss/ICS4U/issues/37 for planned update regarding the searching algorithms.